app/lib/features/space/widgets/space_sections/chats_section.dart

```diff
@@ -44,12 +44,34 @@ class ChatsSection extends ConsumerWidget {
   Widget buildChatsSectionUI(
     BuildContext context,
     WidgetRef ref,
-    List<Convo> chats,
-    bool hasMore,
+    List<String> chats,
   ) {
-    int chatsLimit = (chats.length > limit) ? limit : chats.length;
-    int moreCount = limit > chats.length ? limit - chats.length : 0;
-    bool isShowSeeAllButton = (chats.length > chatsLimit) || hasMore;
```

Comment:
this was another source for a bug, where we'd attempt to figure out the specifics of when to show more items, we'd overestimate (because we didn't take into account whether any were actually resolved remotely).

The basic logic goes like this:
1. show up to `limit` items (usually 3)
2. fill up with known chats
3. if there are more, cut at 3, show the "See all" button
4. if there are spaces left, check whether we have some from remote
5. and if so, fill them up. but also take into account that the total should stay at 3 and only if there are more do we want to show the `Show all` button.

The new check is a lot more expressive and checks the actual remote data we can fetch. And updates accordingly. So we might only show 2 for a moment until we learn that there is another few remote ones.

This should also properly fix the range-errors we still have seen some times if we knew about further rooms but weren't able to load them from remote. Because we disregard that number entirely now.